DataFormats/L1TCalorimeterPhase2/interface/DigitizedTowerCorrelator.h

```diff
@@ -0,0 +1,109 @@
+    }
+
+    ap_uint<4> digitizeHoE(ap_uint<4> hoe) { return (ap_uint<4>)hoe; }
+
+    // To-do: FB not implemented yet
+    ap_uint<2> digitizeFB(ap_uint<2> fb) { return (ap_uint<2>)fb; }
+
+  public:
+    DigitizedTowerCorrelator() { towerData = 0x0; }
+
+    DigitizedTowerCorrelator(ap_uint<16> data) { towerData = data; }
+
+    // Constructor from digitized inputs
+    DigitizedTowerCorrelator(ap_uint<10> et,
+                             ap_uint<4> hoe,
+                             ap_uint<2> fb,
+                             unsigned int indexCard,
+                             unsigned int indexFiber,
+                             unsigned int indexTower,
+                             bool fullyDigitizedInputs) {
+      (void)fullyDigitizedInputs;
+      towerData = ((ap_uint<16>)et) | (((ap_uint<16>)hoe) << 10) | (((ap_uint<16>)fb) << 14);
+      idxCard = indexCard;
+      idxFiber = indexFiber;
+      idxTower = indexTower;
+      assert(hasValidIndices());
+    }
+
+    // Constructor from float inputs
+    DigitizedTowerCorrelator(float et_f,
+                             ap_uint<4> hoe,
+                             ap_uint<2> fb,
+                             unsigned int indexCard,
+                             unsigned int indexFiber,
+                             unsigned int indexTower) {
+      towerData = ((ap_uint<16>)digitizeEt(et_f)) | (((ap_uint<16>)hoe) << 10) | (((ap_uint<16>)fb) << 14);
+      idxCard = indexCard;
+      idxFiber = indexFiber;
+      idxTower = indexTower;
+      assert(hasValidIndices());
+    }
+
+    ap_uint<16> data() const { return towerData; }
+
+    // Other getters
+    float etLSB() const { return LSB_ET; }
+    ap_uint<10> et() const { return (towerData & 0x3FF); }        // ten 1's = 0x3FF
+    ap_uint<4> hoe() const { return ((towerData >> 10) & 0xF); }  // four 1's= 0xF
+    ap_uint<2> fb() const { return ((towerData >> 14) & 0x3); }   // two 1's = 0x3
+    float etFloat() const { return et() * etLSB(); }
+    unsigned int cardNumber() const { return idxCard; }    // GCT card number
+    unsigned int fiberNumber() const { return idxFiber; }  // fiber number in card (hardware convention)
+    unsigned int towerNumber() const { return idxTower; }  // tower number in fiber (hardware convention)
+
+    // Prints
+    void print(const std::string location);
+    void printEt(void) const { std::cout << "et: " << std::bitset<10>{et()} << std::endl; };
+    void printEtFloat(void) const { std::cout << "et float: " << etFloat() << std::endl; };
+    void printHoE(void) const { std::cout << "hoe: " << std::bitset<4>{hoe()} << std::endl; };
+    void printFB(void) const { std::cout << "fb: " << std::bitset<2>{fb()} << std::endl; };
+    void printInfo(void) const {
+      std::cout << "GCT card: " << cardNumber() << ", fiber number: " << fiberNumber()
+                << ", tower number: " << towerNumber() << ", energy (GeV): " << etFloat() << std::endl;
+    };
```

Comment:
Remove these

Reply:
Removed as of [e7e753d](https://github.com/cms-l1t-offline/cmssw/pull/1069/commits/e7e753df69e7aff845a3b3dfbaed9a06c9fdaec6)